src/dft/backends/mklcpu/mklcpu_helpers.hpp

```diff
@@ -0,0 +1,201 @@
+/*******************************************************************************
+* Copyright 2022 Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _ONEMKL_DFT_SRC_MKLCPU_HELPERS_HPP_
+#define _ONEMKL_DFT_SRC_MKLCPU_HELPERS_HPP_
+
+#include "oneapi/mkl/detail/exceptions.hpp"
+#include "oneapi/mkl/dft/detail/types_impl.hpp"
+
+// MKLCPU header
+#include "mkl_dfti.h"
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+namespace mklcpu {
+namespace detail {
+
+template<typename T> struct is_buffer : std::false_type {};
+template<> struct is_buffer<sycl::buffer<float>> : std::true_type {};
+template<> struct is_buffer<sycl::buffer<std::complex<float>>> : std::true_type {};
+template<> struct is_buffer<sycl::buffer<double>> : std::true_type {};
+template<> struct is_buffer<sycl::buffer<std::complex<double>>> : std::true_type {};
+
+template<typename T> struct is_usm : std::false_type {};
+template<> struct is_usm<float> : std::true_type {};
+template<> struct is_usm<std::complex<float>> : std::true_type {};
+template<> struct is_usm<double> : std::true_type {};
+template<> struct is_usm<std::complex<double>> : std::true_type {};
+
+using mklcpu_desc_t = DFTI_DESCRIPTOR_HANDLE;
+using commit_t = dft::detail::commit_impl;
+template< class T>
+inline constexpr bool is_buffer_v = is_buffer<T>::value;
+template< class T>
+inline constexpr bool is_usm_v = is_usm<T>::value;
```

Comment:
```suggestion
using is_buffer_v = is_buffer<T>::value;
template< class T>
using is_usm_v = is_usm<T>::value;
```
would be more typical.

Reply:
My comment here is incorrect. Please ignore it.